### ISSUE TITLE
Point number limit in PCA decomposition

### ISSUE THREAD

**Issue by hugo-pires** [DistrictDataLabs/yellowbrick#920]
Hello

Is there any limit to the number of points in PCADecomposition plot?

Thank you

**Comment by lwgray** [comment]
hi @hugo-pires Thank you for using Yellowbrick.  I don’t think there is an upper-limit but i will verify this with my colleagues.  Are you having problems with the PCADecomposition visualizer?

**Comment by hugo-pires** [comment]
Thank you and congratulations for your work.

The questions is that I am projecting an aprox 5000 examples dataframe, but I'm getting an aprox 50 points. Probably they are overlaid, but I would like to doublecheck. Or use some alpha parameter.

**Comment by rebeccabilbro** [comment]
Hi @hugo-pires &mdash; thanks for checking out Yellowbrick! There probably is an upper limit for PCA, but 5k samples isn't even close to it! So I agree it's likely that the points are merely being overlaid. You can test this out using the `alpha` parameter, as so:

```python
from sklearn.datasets import make_classification
from yellowbrick.features.pca import PCADecomposition

X, y = make_classification(
    n_samples=5000, n_features=200, n_informative=2, n_redundant=2
)

visualizer = PCADecomposition(alpha=.25)
visualizer.fit_transform(X, y)
visualizer.poof()
``` 

The default `alpha` is .75 (check out [our `PCADecomposition` docs](https://www.scikit-yb.org/en/develop/api/features/pca.html#module-yellowbrick.features.pca) for more on our defaults) and ranges between 0 and 1; as it decreases, the points will become more translucent.

`alpha=.75`:
![image](https://user-images.githubusercontent.com/8760385/60983517-ad568700-a2ff-11e9-9d09-49406ee4a33c.png)

`alpha=.25`:
![image](https://user-images.githubusercontent.com/8760385/60983532-b2b3d180-a2ff-11e9-80d3-db8228141b3d.png)

**Comment by lwgray** [comment]
@hugo-pires @rebeccabilbro is absolutely right that you should be able to plot 5000 points and her suggestion on altering alpha is right on.  I wanted to add a few things.

1.  We are actively working on the PCA visualizer and the next version will have advanced support for opacity
2.  You can use `proj_features=True` to create a biplot which would help you interpret what's going on in the decomposition
3.  Alternatively you can use a Manifold - non linear decomposition to see if it's just a linear effect
4. TSNE or isomap might tell you a lot about what's going on in the higher dimensional space

**Comment by hugo-pires** [comment]
Thank you @lwgray and @rebeccabilbro . It would also be nice to have auto colors to discrete targets, based on the levels of the label.

**Comment by lwgray** [comment]
@hugo-pires i think that #476 and #874 will address that request.
